File: frontend/mobile/lib/core/api_response.dart
class ApiResponse<T> {
  final bool success;
  final String message;
  final T? data;
  final dynamic errors;
  final String timestamp;
  final String? requestId;

  ApiResponse({
    required this.success,
    required this.message,
    this.data,
    this.errors,
    required this.timestamp,
    this.requestId,
  });

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic)? fromJsonT,
  ) {
    return ApiResponse(
      success: json['success'] as bool,
      message: json['message'] as String,
      data: json['data'] != null && fromJsonT != null
          ? fromJsonT(json['data'])
          : json['data'] as T?,
      errors: json['errors'],
      timestamp: json['timestamp'] as String,
      requestId: json['request_id'] as String?,
    );
  }
}

File: backend/app/Modules/Core/Services/ApiResponse.php
<?php

declare(strict_types=1);

namespace App\Modules\Core\Services;

use Illuminate\Http\JsonResponse;

final class ApiResponse
{
    public static function success(
        mixed $data = null,
        string $message = 'تمت العملية بنجاح',
        int $code = 200,
    ): JsonResponse {
        return self::build(true, $message, $data, null, $code);
    }

    public static function error(
        string $message = 'حدث خطأ',
        mixed $errors = null,
        int $code = 400,
    ): JsonResponse {
        return self::build(false, $message, null, $errors, $code);
    }

    private static function build(
        bool $success,
        string $message,
        mixed $data,
        mixed $errors,
        int $code,
    ): JsonResponse {
        return response()->json([
            'success' => $success,
            'message' => $message,
            'data' => $data,
            'errors' => $errors,
            'timestamp' => now()->toIso8601String(),
            'request_id' => request()->header('X-Request-Id'),
        ], $code);
    }
}
